ddtrace/tracer/option.go

```diff
@@ -84,8 +89,8 @@ type config struct {
 // StartOption represents a function that can be provided as a parameter to Start.
 type StartOption func(*config)
 
-// defaults sets the default values for a config.
-func defaults(c *config) {
+func newConfig(opts ...StartOption) *config {
```

Comment:
I don't know if I like this huge function more. IMHO we could instead have 2 smaller ones:

```
c := new(config)
defaults(c)
for _, fn := range opts {
    fn(c)
}
c.validate() // could return the error for Start in v2
```
I don't feel strongly about this. In Go sometimes there are long, verbose functions. In JavaScript, smaller ones are always preferred, even if never re-used.

Just some food for thought, you can leave this.

Reply:
I would prefer a smaller function too.
I looked for a logical place to break the function up, but didn't see an obvious one. The old `defaults` wasn't setting defaults, but picking up env variables and other things. 

I'll give this another look and see if I can make it better. Something like your two-function idea will probably work nicely.